internal/updates/updates.go

```diff
@@ -108,6 +110,16 @@ func Update(ctx context.Context, currentVersion, artifactArch string, timeout in
 // InstallVersion installs a specific version of the CLI
 // returns the path to the installed binary
 func InstallVersion(ctx context.Context, desiredVersion, artifactArch string, timeout int) (string, error) {
+	mutex := concurrency.NewIPMutex()
+	err := mutex.TryLock(ctx, func(attempt int) {
+		log.From(ctx).PrintfStyled(styles.DimmedItalic, "Failed to acquire lock (attempt %d). Retrying...", attempt)
+	})
+	if err != nil {
+		return "", err
+	}
```

Comment:
```suggestion
A real nit but it might be cleaner to have

```
l := locks.CLIUpgradeLock()
for attempt, err := l.Acquire({ Timeout: time.Minute }) {
   log.From(ctx).PrintfStyled(styles.DimmedItalic, "Failed to acquire lock (attempt %d). Retrying...", attempt)
  if err { return err }
}
defer l.Release()
```